pilot/pkg/model/authentication.go

```diff
@@ -88,27 +95,60 @@ func (policy *AuthenticationPolicies) addRequestAuthentication(configs []Config)
 	}
 }
 
+func (policy *AuthenticationPolicies) addPeerAuthentication(configs []Config) {
+	for _, config := range configs {
+		policy.peerAuthentications[config.Namespace] =
+			append(policy.peerAuthentications[config.Namespace], config)
+	}
+}
+
 // GetJwtPoliciesForWorkload returns a list of JWT policies matching to labels.
 func (policy *AuthenticationPolicies) GetJwtPoliciesForWorkload(namespace string,
+	workloadLabels labels.Collection) []*Config {
+	extractor := func(cfg *Config) map[string]string {
+		return cfg.Spec.(*v1beta1.RequestAuthentication).GetSelector().GetMatchLabels()
```

Comment:
We should be able to just do type switch of the opaque config, and then get label from config spec.

this collapse the oneliner funtion into a consolidated place, easier to follow.

Reply:
Done